nuxeo-drive-client/nxdrive/engine/engine.py

```diff
@@ -1005,3 +1009,29 @@ def fetch_all_users(self):
         except Exception as e:
             log.exception(e)
         return
+
+    def refresh_user(self, user_id):
```

Comment:
I understand and is ok with me. Generally, the rule about performance optimization is don't do it until you know there is a performance problem.
One could write a test to compare the avg. time for a simple SQLite lookup by the primary key (indexed), vs a dictionary lookup. Or, lookup some performance figures, e.g. https://www.sqlite.org/speed.html: 
5000 selects (w/ an index) 
SQLite 2.7.6:	   1.121s
which amounts to 0.23ms/query. A call to GetFileSystemItem (which would trigger the SQLite lookup) takes 550ms (from the log). This means this cache squeezes a 0.035% performance improvement (not even considering all the other HTTP requests and processing which comprise syncing a file).
I would take simpler code any day :-)